### DISCUSSION TITLE
Add feature to download HLS video from M3U8 files

### DISCUSSION THREAD

**Discussion by trimechee** [setvisible/DownZemAll#191] [Ideas]
Hi, sometimes I encounter problems in downloading when I playback the temporary hls video downloaded by the famous download videos extensions like:

https://github.com/xifangczy/cat-catch

and

https://github.com/chandler-stimson/live-stream-downloader

this extension is not a downloader but can also detect m3u8 links of embedded hls videos for preview:

https://github.com/andy-portmen/open-in-vlc

and also these extensions are able to give us the m3u8 links of the embedded hls videos:

https://github.com/puemos/hls-downloader

https://fetchv.net/

https://github.com/54ac/stream-detector

https://cococut.net/

http://sniffertv.com/docs/
https://sniffertv.com/docs/

https://www.downloadhelper.net/

https://fdown.net/

https://9xbuddy.com/fr
https://9xbuddy.online/fr

https://mediatools.rndnm.com/howTo
https://mediatools.cc/hlsDownloader

https://www.startpage24.com/

https://m3u8.dev/
https://download.m3u8.dev/
https://beta.m3u8.dev/
https://m3u8play.dev/

i.e. the problems I encounter sometimes are the download does not finish or is deleted, web browser that does not close at the end of the download despite the auto close browser when donwload finish option is enabled....

I had the idea to use external downloaders to copy paste the m3u8 link given by these great extensions and put them in external donwload managers, like FDM, media-donwloader, jdownloader....but either it doesn't work and these download managers are unable to download hls m3u8 videos, or there is no sequential download and we can't playback temporary video because it's full of small fragments...

until this night, I decide to install IDA a famous Russian download manager because it's compatible with my indows 7 os 32 bits, and there the dream becomes true alhamdouilAllah, I copy the m3u8 link given by the extensions and I paste it into IDA: Internet Download Accelerator which can download the hls video sequentially and display the temporary video and I can playback temporary video! I set the number of threads = 1, and I enabled the option from the graphical interface "close IDA when download finish".....

so we hope please ArrowDL  will also be able to download hls videos if we copy and paste the m3u8 links given by the video downloaders extensions, so ArrowDL will no longer need to develop a custom media detection extension, and we could download hls video and playback temporary video whose download should be sequential i.e. without displaying lots of small hls video fragments in the download folder in order to be able to playback temporary video, thank's!

**Comment by setvisible** [comment]
Same anwser as #160 _Add feature to download M3U8 files_

This request could be a proposition to be implemented in yt-dlp directly.

I'll close it.

Regards
